### PR TITLE
Merge FoodList branch

### LINKED ISSUE #9: As a person on a diet, I can input the food that I ate
...so that I can keep track of what I have eaten.

### LINKED ISSUE #10: As a clumsy person, I can delete things that I have inputted wrongly
...so that my diet tracker remains accurate.

### LINKED ISSUE #13: Add support for viewing of food items that user has consumed to FoodList class

### PR DESCRIPTION
**Pull request by 0xEljh**
Add FoodList and core subdependencies in the list packge.

Can now create a foodlist that stores food entries with food details and portion sizes.
These can be added, deleted, and printed. List can also be cleared.

> Implementations:
> - Foodlist is a wrapper around the ArrayList representation of the data. Supports modification of data.
> 
> - FoodListManager is to streamline the FoodList class. Handles modification beyond arraylist methods. Adds potential for FP.
> 
> - FoodEntry is a data class to support the storage of data.

Fixes #13 
Fixes #10
Fixes #9

### CODE REVIEW

File: src/main/java/seedu/duke/list/FoodEntry.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            int carbohydrate, int protein, int fat) {
+        this.portionSize = portionSize;
+        this.food = new Food(name, calorie, carbohydrate, protein, fat);
+    }
+
+    public Food getFood() {
+        return food;
+    }
+
+    public int portionSize() {
```
Comment (HengFuYuen) [inline src/main/java/seedu/duke/list/FoodEntry.java]: Perhaps you can consider changing the method name to getPortionSize.

File: src/main/java/seedu/duke/list/FoodEntry.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return portionSize;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s -- (%s)", food.toString(), portionSize);
+    }
+
+}
```
Comment (HengFuYuen) [inline src/main/java/seedu/duke/list/FoodEntry.java]: Perhaps you might want to leave a blank line at the end of the file.

File: src/main/java/seedu/duke/list/FoodList.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    public String addFood(int portionSize, String name) throws FoodNotFoundException {
+        throw new FoodNotFoundException();
+    }
+
+
+    public String delete(int index) throws IndexOutOfBoundsException {
+        try {
+            return FoodListManager.deleteEntry(foodEntries, index).toString();
+        } catch (IndexOutOfBoundsException e) {
+            throw e;
```
Comment (HengFuYuen) [inline src/main/java/seedu/duke/list/FoodList.java:61]: Maybe you can consider including an error message.

File: src/main/java/seedu/duke/list/FoodListManager.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        });
+        return foods;
+    }
+
+    protected static Food deleteEntry(ArrayList<FoodEntry> list, int index) throws IndexOutOfBoundsException {
+        int indexToDelete = index - 1;
+        try {
+            return list.remove(indexToDelete).getFood();
+        } catch (IndexOutOfBoundsException e) {
+            throw e;
```
Comment (HengFuYuen) [inline src/main/java/seedu/duke/list/FoodListManager.java:38]: Maybe you can consider including an error message.

File: src/main/java/seedu/duke/list/FoodListManager.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        for (int i = 1; i <= list.size(); i++) {
+            FoodEntry task = list.get(i - 1);
+            listString += i + ". "
+                    + task.toString() + "\n";
+        }
+        return listString;
+    }
+
+    protected static ArrayList<Food> listToFoods(ArrayList<FoodEntry> list) {
```
Comment (HengFuYuen) [inline src/main/java/seedu/duke/list/FoodListManager.java:33]: Maybe you can consider adding JavaDoc comments to some of the methods in this class.

File: src/test/java/seedu/duke/FoodListTest.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        Food food = new Food("Kobe Beef", 480,50,40,30);
+        FoodList foodList = new FoodList();
+        
+        System.out.println(foodList.addFood(3, food));
+        System.out.println(foodList.addFood(2, "Sashimi", 100, 0, 30, 10));
+        System.out.println(foodList);
+        System.out.println(foodList.delete(1));
+        System.out.println(foodList);
+    }
+}
```
Comment (HengFuYuen) [inline src/test/java/seedu/duke/FoodListTest.java:19]: Perhaps you would like to split them into different tests using assertEquals, assertTrue, etc for testing and using a three-part naming style for each method e.g. methodUnderTest_inputGiven_expectedOutput

**Review by HengFuYuen — COMMENTED** [review]
Perhaps some minor changes are needed to improve code quality and testing.

**Comment by 0xEljh** [comment]
Changes to testing will be made in a new branch since this branch already has its core functionality. (removed fix #26) 

Plan to further streamline these features by adding more FP support and reducing repetition.

**Review by HengFuYuen — APPROVED** [review]
Current commits LGTM.

**Pull request merged**